src/io/data.jl

```diff
@@ -231,13 +231,49 @@ function read_table(filenames::AbstractVector)
     for i in 2:length(filenames)
         filename = filenames[i]
         tmp = read_table(filename)
+
+        # Check to see if either table has columns that the other does not.
+        original_table_extra_cols = setdiff(propertynames(table), propertynames(tmp))
+        new_table_extra_cols = setdiff(propertynames(tmp), propertynames(table))
+
+        # Add extra columns to `tmp`
+        for extra_col in original_table_extra_cols
+            if hasmethod(get_default_column_value, Tuple{Val{extra_col}})
+                tmp[!, extra_col] .= get_default_column_value(Val(extra_col))
+            else
+                @warn "No column $(extra_col) defined in file $(filename), removing that column from both tables. Consider defining a default value using get_default_column_value"
+                select!(table, Not(extra_col))
+            end
+        end
+
+        # Add extra columns to `table`
+        for extra_col in new_table_extra_cols
+            if hasmethod(get_default_column_value, Tuple{Val{extra_col}})
+                table[!, extra_col] .= get_default_column_value(Val(extra_col))
+            else
+                @warn "No column $(extra_col) defined in the table that file $(filename) is being added to, removing that column from both tables. Consider defining a default value using get_default_column_value"
```

Comment:
This all looks good, but I'm a bit confused on what lines 244 and 254 (with the warn function) in the code are actually doing, are they technically comments?

Reply:
Good question - they are logging statements.  When E4ST.jl runs, it has the option of logging to a log file, or if you don't specify a log file, it will log to the REPL (i.e. the command line).  Any time that you see the following macros, it is logging.
* `@debug`
* `@info`
* `@warn`
* `@error`

Log files look something like this:
![image](https://github.com/e4st-dev/E4ST.jl/assets/44298283/262ed8f4-4d66-45d3-8f82-d7ce2fdb2808)

It is handy because it shows you where in the code it came from, at what time, and gives the message.

Reply:
Ahh okay that makes sense, that's super useful to know, thank you!